### PR TITLE
PR feedback on recent work to sync solutions consistently

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Remaining feedback on https://github.com/dotnet/roslyn/pull/72860

### CODE REVIEW

File: src/Workspaces/Core/Portable/SourceGeneration/IRemoteSourceGenerationService.cs [hunk 1]
```diff
@@ -27,24 +27,22 @@ internal interface IRemoteSourceGenerationService
...
     /// </summary>
-    /// <remarks>
-    /// Should only be called by the "RegularCompilationTracker", and should only return data from its view of the
-    /// world.  Not from the view of a "GeneratedFileReplacingCompilationTracker".
-    /// </remarks>
-    ValueTask<ImmutableArray<RegularCompilationTrackerSourceGenerationInfo>> GetRegularCompilationTrackerSourceGenerationInfoAsync(
-        Checksum solutionChecksum, ProjectId projectId, CancellationToken cancellationToken);
+    /// <param name="withFrozenSourceGeneratedDocuments">Controls if the caller wants frozen source generator documents
+    /// included in the result, or if only the most underlying generated documents (produced by the real compiler <see
+    /// cref="GeneratorDriver"/> shoudl be included.</param>
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/SourceGeneration/IRemoteSourceGenerationService.cs:32]:
```suggestion
    /// cref="GeneratorDriver"/> should be included.</param>
```

File: src/Workspaces/Core/Portable/SourceGeneration/IRemoteSourceGenerationService.cs [hunk 2]
```diff
@@ -27,24 +27,22 @@ internal interface IRemoteSourceGenerationService
...
-    /// <remarks>
-    /// Should only be called by the "RegularCompilationTracker", and should only return data from its view of the
-    /// world.  Not from the view of a "GeneratedFileReplacingCompilationTracker".
-    /// </remarks>
-    ValueTask<ImmutableArray<RegularCompilationTrackerSourceGenerationInfo>> GetRegularCompilationTrackerSourceGenerationInfoAsync(
-        Checksum solutionChecksum, ProjectId projectId, CancellationToken cancellationToken);
+    /// <param name="withFrozenSourceGeneratedDocuments">Controls if the caller wants frozen source generator documents
+    /// included in the result, or if only the most underlying generated documents (produced by the real compiler <see
+    /// cref="GeneratorDriver"/> should be included.</param>
+    ValueTask<ImmutableArray<RegularCompilationTrackerSourceGenerationInfo>> GetSourceGenerationInfoAsync(
```
Comment (ToddGrun) [inline src/Workspaces/Core/Portable/SourceGeneration/IRemoteSourceGenerationService.cs:33]: Maybe rename RegularCompilationTrackerSourceGenerationInfo to something like SourceGenerationInfo

Comment (CyrusNajmabadi) [reply src/Workspaces/Core/Portable/SourceGeneration/IRemoteSourceGenerationService.cs:33]: oh yeah.  missed this.

**Comment by CyrusNajmabadi** [comment]
@jasonmalinowski as well so he can see the cleanup done post merge.

**Review by ToddGrun — APPROVED** [review]

**Pull request merged**